kornia/utils/misc.py

```diff
@@ -23,7 +23,7 @@ def eye_like(n: int, input: torch.Tensor, shared_memory: bool = False) -> torch.
     if len(input.shape) < 1:
         raise AssertionError(input.shape)
 
-    identity = torch.eye(n, device=input.device, dtype=input.dtype)
+    identity = torch.eye(n, device=input.device).type(input.dtype)
```

Comment:
why ?

Reply:
@edgarriba `torch.eye` does not support `bfloat16`